cmd/cloudinit/cloudinit.go

```diff
@@ -230,6 +231,7 @@ func saveCloudConfig() error {
 	}
 
 	userData := string(userDataBytes)
+	userDataBytes = substituteVars(userData, metadata)
```

Comment:
Hi @imikushin, it's exactly what i want though! :) userData does not need to be changed. It is just used for `IsScript` / `IsCompose` and then discarded anyway. It is userDataBytes that needs to be assigned the new value.